src/registry.rs

```diff
@@ -95,27 +95,28 @@ impl Registry {
         self.r.write().unwrap().unregister(c)
     }
 
-    pub fn get_core(&self) -> RwLockReadGuard<RegistryCore> {
-        self.r.read().unwrap()
-    }
-}
+    pub fn scrap(&self, writer: &mut Write, encoder: &Encoder) -> Result<usize> {
```

Comment:
refer prometheus golang client, support gather function to return all metrics, and then use encoder to encode metrics to writer later.